### ISSUE TITLE
problem with wsb2file output

### ISSUE THREAD

**Issue by samspo** [danny0838/PyWebScrapBook#45]
MACOS 10.15.7 
webscrapbook 0.40.0

Hello,

I tried convert a small (test) scrapbook from ScrapbookX with wsb convert sb2wsb --> it works perfectly
Then, I wanted to convert the scrapbook from "folder" to "files" --> I used wsb convert wsb2file ( I suppose it should convert my folder into html files, am I right?)
At the end, I have the folder (my structure) but in that, instead of html files, I have some htd folders. Is it the good results? Am I missing something?

<img width="525" alt="Capture d’écran, le 2021-04-18 à 18 02 33" src="https://user-images.githubusercontent.com/1280210/115162304-be096480-a070-11eb-9f82-c4424f5ce467.png">

```wsb convert wsb2file /Users/sam/Documents/Test/ST8 /Users/sam/Documents/Test/ST9
INFO: conversion mode: WebScrapBook --> hierarchial files
INFO: input directory: /Users/sam/Documents/Test/ST8
INFO: output directory: /Users/sam/Documents/Test/ST9
INFO: book ID: ""
INFO: prefix: True
INFO: 
INFO: Exporting "20210418104536" to "1-MyF"...
INFO: Exporting "20210418104524" to "1-MyF/1-Diffs · danny0838_webscrapbook Wiki · GitHub.htd"...
INFO: Exporting "20210418104603" to "1-MyF/2-Prêt à « sanctionner », Emmanuel Macron veut « définir de claires lignes rouges avec la Russie ».htd"...
INFO: Exporting "20210418104534" to "1-MyF/3-Screenshots · danny0838_webscrapbook Wiki · GitHub.htd"...
INFO: Exporting "20210418174444" to "2-AnotherF"...
INFO: Exporting "20210418174640" to "2-AnotherF/1-I think I figured out what DFV knows... and it’s pretty simple. _ Superstonk.htd"...
INFO: Done.
INFO: Time spent: 0.5495898723602295 seconds.```

**Comment by danny0838** [comment]
This is intended. WSB allows any item to have child items. If you have an item `myitem` with child item `abc`, files of `myitem` is output to `./myitem.htd/*` and child items are output to `./myitem/*`, so that they won't conflict with each other.

This also makes data structure more clear—a directory named `*.htd` stands for an item, while an unsuffixed directory is a wrapper of child items.

**Comment by samspo** [comment]
Thanks a lot for you your quick response, Danny.
No possibility to export the folders as single html files?

**Comment by danny0838** [comment]
Although it's expected that someone may want a feature to transform between scrapbook folder, HTZ, MAFF, and single HTML file format, it's currently not supported. Convertion between scrapbook folder, HTZ, and MAFF should be relatively easy as they are largely interchangeable, while conversion between those formats and single HTML is much more complicated and may not be implemented so soon.

Also, there are lots of potential similar features that may be requested, such as:
1. A CLI tool to pick some items and convert their data files from a folder into a single HTML.
1. A CLI tool to pick some items and export them as single HTML files.
1. Select some items in the sidebar and convert their data files from a folder into a single HTML.
1. Select some items in the sidebar and export them as single HTML files.
1. Convert data files of each WebScrapBook archive file from a folder into a single HTML.
1. Convert each WebScrapBook archive file into a single HTML.

and even more...

They all look similar and somehow interchangeable, but unfortunately we don't have the resource to support all variations of them in the short run, and a prioritization is required. Currently we tend to go on the first approach, as it seems more reusable and can be used to achieve other requirements with a combination of other utilities (for example, select wanted items, copy them into a new empty scrapbook, convert items in that scrapbook into single HTML file, and "export" them using `wsb convert wsb2file`). Any idea and feedback is welcome, though.

**Comment by samspo** [comment]
Hello Danny,
Thanks for the follow up.
Yes. First approach would be useful.
I managed to do it with scrapbook x and the converter.
Thx again for all the efforts you put in these tools,

**Comment by danny0838** [comment]
We have added `items` converter that can convert items filtered by type and/or ID into other format.

Currently only convertion between folder, htz, and maff is supported. Although convertion between these formats and "single file" is not lossless and less encouraged, and it may somehow be more convenient to just convert to single file during `wsb2file` converter, we decided to implement single file convertion for `items` converter first as is may still be useful in certain cases due to its selectivity and flexibility.
